app/controllers/irregular_incomes_controller.rb

```diff
@@ -2,22 +2,20 @@ class IrregularIncomesController < ApplicationController
   before_action :load_assessment
 
   def create
-    if creation_service.success?
+    json_validator = JsonValidator.new("irregular_incomes", irregular_income_params)
```

Comment:
@starswan What do you think of using a `before_action` callback for validating json instead of using if-else?

Reply:
It still leaves some duplication, which I am tackling in LEP-57